test/test.ts

```diff
@@ -90,13 +89,19 @@ describe('cloud functions api', () => {
 });
 
 describe('end to end', () => {
-  it('should work together end to end', async () => {
+  it('should deploy end to end', async () => {
     const scopes = [mockUploadUrl(), mockUpload(), mockDeploy(), mockPoll()];
     const projectId = 'el-gato';
     const deployer = new gcx.Deployer({name, targetDir, projectId});
     await deployer.deploy();
     scopes.forEach(s => s.done());
   });
+
+  it('should call end to end', async () => {
+    mockUploadUrl(), mockUpload(), mockDeploy(), mockPoll();
+    const c = new gcx.Caller();
+    c.call(name);
```

Comment:
It would be great to nock the calling endpoint, and assert that the response is what we expect.